engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableNegativeZeroTest.java

```diff
@@ -0,0 +1,459 @@
+import io.deephaven.api.agg.spec.AggSpecPercentile;
+import io.deephaven.api.agg.spec.AggSpecSortedFirst;
+import io.deephaven.api.agg.spec.AggSpecSortedLast;
+import io.deephaven.api.agg.spec.AggSpecStd;
+import io.deephaven.api.agg.spec.AggSpecSum;
+import io.deephaven.api.agg.spec.AggSpecTDigest;
+import io.deephaven.api.agg.spec.AggSpecUnique;
+import io.deephaven.api.agg.spec.AggSpecVar;
+import io.deephaven.api.agg.spec.AggSpecWAvg;
+import io.deephaven.api.agg.spec.AggSpecWSum;
+import io.deephaven.api.object.UnionObject;
+import io.deephaven.chunk.util.pools.ChunkPoolReleaseTracking;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.testutil.junit4.EngineCleanup;
+import io.deephaven.test.types.OutOfBandTest;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static io.deephaven.engine.testutil.TstUtils.assertTableEquals;
+import static io.deephaven.engine.util.TableTools.doubleCol;
+import static io.deephaven.engine.util.TableTools.floatCol;
+import static io.deephaven.engine.util.TableTools.intCol;
+import static io.deephaven.engine.util.TableTools.longCol;
+import static io.deephaven.engine.util.TableTools.newTable;
+
+@Category(OutOfBandTest.class)
+public class QueryTableNegativeZeroTest {
+
+    @Rule
+    public final EngineCleanup base = new EngineCleanup();
+
+    @Before
+    public void setUp() throws Exception {
+        ChunkPoolReleaseTracking.enableStrict();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ChunkPoolReleaseTracking.checkAndDisable();
+    }
+
+    @Test
+    public void testDhc3768_float_join_key() {
+        final Table t1 = newTable(floatCol("KeyNeg", -0.0f), floatCol("Neg", -0.0f));
+        final Table t2 = newTable(floatCol("KeyPos", 0.0f), floatCol("Pos", 0.0f));
+        for (Table out : List.of(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key;
+            try (final RowSet.Iterator iterator = out.getRowSet().iterator()) {
+                key = iterator.nextLong();
+            }
+            floatToBitsEquals(out, "KeyNeg", key, -0.0f);
+            floatToBitsEquals(out, "KeyPos", key, 0.0f);
+            floatToBitsEquals(out, "Neg", key, -0.0f);
+            floatToBitsEquals(out, "Pos", key, 0.0f);
+        }
+    }
+
+    @Test
+    public void testDhc3768_double_join_key() {
```

Comment:
rangeJoin and multiJoin?

Reply:
Added multijoin. Spent a little time on rangeJoin, didn't quite grok the correct incantations.